### PR TITLE
Disable value validation for Decimal256 case

### LINKED ISSUE #2233: Disable value validation for Decimal256 case
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

As Decimal128 case, we also need to disable value validation for Decimal256 case for C++ integration test.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #2233.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by viirya** [comment]
cc @alamb

File: integration-testing/src/lib.rs [hunk 1]
```diff
@@ -618,6 +618,10 @@ fn array_from_json(
         }
         DataType::Decimal256(precision, scale) => {
             let mut b = Decimal256Builder::new(json_col.count, *precision, *scale);
+            // C++ interop tests involve incompatible decimal values
```
Comment (liukun4515) [inline integration-testing/src/lib.rs:621]: @viirya 
Hi java or c++ version, Does the decimal builder or the decimal array has the validation for the input value?
If there is no validation, how to make sure all of the element of the decimal array are valid?

Comment (viirya) [reply integration-testing/src/lib.rs:621]: Like Decimal128 case #1766, C++ doesn't perform validation in Decimal builders. It only does validation when doing a full validating ArrayData.

**Review by alamb — APPROVED** [review]
Thanks @viirya 

I wonder if we should file a follow on ticket to track the incompatible value in the C++ integration test? Something doesn't seem right there -- if the integration test is using incompatible values perhaps someone should fix the C++ test

(btw we can see the integration test failing on arrow mater here: https://github.com/apache/arrow-rs/runs/7588317417?check_suite_focus=true and it passes with your PR 🎉 )

**Comment by viirya** [comment]
@alamb Yea, let me create a ticket for the C++ integration test.

**Pull request merged**